File: leetcode/rust-leetcode/src/1768.merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let w1 = word1.into_bytes();
        let w2 = word2.into_bytes();
        let mut ans:String = String::new();
        let l1 = w1.len();
        let l2 = w2.len();
        for i in 0..l1.max(l2) {
            if i < l1 {
                ans.push(w1[i] as char);
            }
            if i < l2 {
                ans.push(w2[i] as char);
            }
        }
        ans
    }
}

impl Solution2 {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut chars1 = word1.chars();
        let mut chars2 = word2.chars();
        let mut ans = String::with_capacity(word1.len() + word2.len());
        for _ in 0..word1.len().min(word2.len()) {
            ans.push(chars1.next().unwrap());
            ans.push(chars2.next().unwrap());
        }
        ans.extend(chars1);
        ans.extend(chars2);
        ans
    }
}

File: leetcode/python/1768.merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i1, i2 = 0, 0
        ans = ""
        while i1 < len(word1) and i2 < len(word2):
            ans += word1[i1] + word2[i2]
            i1 += 1
            i2 += 1
        if i1 < len(word1):
            ans += word1[i1:]
        if i2 < len(word2):
            ans += word2[i2:]
        return ans

""" """

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = 0
        ans = ""
        while i < len(word1) or i < len(word2):
            if i < len(word1):
                ans += word1[i]
            if i < len(word2):
                ans += word2[i]
            i += 1
        return ans

